tests/shared/drive-constraints.spec.js

```diff
@@ -771,4 +772,241 @@ describe('Shared: DriveConstraints', function() {
 
   });
 
+  describe('.getDriveImageCompatibilityStatuses', function() {
+
+    beforeEach(function() {
+      if (process.platform === 'win32') {
+        this.mountpoint = 'E:';
+        this.separator = '\\';
+      } else {
+        this.mountpoint = '/mnt/foo';
+        this.separator = '/';
+      }
+
+      this.drive = {
+        device: '/dev/disk2',
+        name: 'My Drive',
+        protected: false,
+        system: false,
+        mountpoints: [
+          {
+            path: this.mountpoint
+          }
+        ],
+        size: 4000000000
+      };
+
+      this.image = {
+        path: path.join(__dirname, 'rpi.img'),
+        size: {
+          original: this.drive.size - 1,
+          final: {
+            estimation: false
+          }
+        }
+      };
+    });
+
+    const expectStatusTypesAndMessagesToBe = (resultList, expectedTuples) => {
+
+      // Sort so that order doesn't matter
+      const expectedTypesSorted = _.sortBy(_.map(expectedTuples, (tuple) => {
+        return constraints.COMPATIBILITY_STATUS_TYPES[tuple[0]];
+      }));
+      const expectedMessagesSorted = _.sortBy(_.map(expectedTuples, (tuple) => {
+        return constraints.COMPATIBILITY_STATUS_MESSAGES[tuple[1]];
+      }));
+      const resultTypesSorted = _.sortBy(_.map(resultList, 'type'));
+      const resultMessagesSorted = _.sortBy(_.map(resultList, 'message'));
```

Comment:
I'm not 100% sure what this does, but does it mean that e.g. `[ ['ERROR', 'B'], ['WARNING', 'A'] ]` would compare as identical to `[ ['ERROR', 'A'], ['WARNING', 'B'] ]` ?  (which IMHO would be wrong)

I dunno how sorting works in JavaScript / NodeJS / Lodash but can you use a 'nested' comparison function, so that it first sorts by `tuple[0]` and then sorts by `tuple[1]` ? So that e.g. `[ ['ERROR', 'Z'], ['WARNING', 'A'], ['ERROR', 'B'] ]` would sort as `[ ['ERROR', 'B'], ['ERROR', 'Z'], ['WARNING', 'A'] ]`